packages/zendesk-adapter/src/config_creator.ts

```diff
@@ -0,0 +1,61 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { BuiltinTypes, ConfigCreator, ElemID, InstanceElement, ObjectType } from '@salto-io/adapter-api'
+import { createDefaultInstanceFromType } from '@salto-io/adapter-utils'
+import { logger } from '@salto-io/logging'
+import { configType } from './config'
+import * as constants from './constants'
+
+const log = logger(module)
+
+const optionsElemId = new ElemID(constants.ZENDESK, 'configOptionsType')
+
+export const optionsType = new ObjectType({
+  elemID: optionsElemId,
+  fields: {
+    enableGuide: { refType: BuiltinTypes.BOOLEAN },
+  },
+})
+
+const isOptionsTypeInstance = (instance: InstanceElement):
+  instance is InstanceElement & { value: { enableGuide?: boolean } } => {
```

Comment:
another type nit (sorry I missed it last time) - if you actually define the supported flags as a `ConfigOptionsType`, then you can `createMatchingObjectType` and ensure these are aligned as some extra safety - 
```suggestion
type ConfigOptionsType = {
  enableGuide?: boolean
}
export const optionsType = createMatchingObjectType<ConfigOptionsType>({
  elemID: optionsElemId,
  fields: {
    enableGuide: { refType: BuiltinTypes.BOOLEAN },
  },
})

const isOptionsTypeInstance = (instance: InstanceElement):
  instance is InstanceElement & { value: ConfigOptionsType } => {
```

Reply:
(or just `ConfigOptions` without `Type`)